contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1416,6 +1416,32 @@ export const ContentNode = new TreeResource({
     });
   },
   setChannelIdOnChange: setChannelIdFromTransactionSource,
+
+  /**
+   * Waits for copying of content nodes to complete for all ids referenced in `ids` array
+   * @param {string[]} ids
+   * @returns {Promise<void>}
+   */
+  waitForCopying(ids) {
+    const observable = Dexie.liveQuery(() => {
+      return this.table.where('id').anyOf(ids).filter(node => !node[COPYING_FLAG]).toArray();
+    })
+
+    return new Promise((resolve, reject) => {
+      const subscription = observable.subscribe({
+        next(result) {
+          if (result.length === ids.length) {
+            subscription.unsubscribe();
+            resolve();
+          }
+        },
+        error() {
+          subscription.unsubscribe();
+          reject();
+        }
+      });
+    });
+  }
```

Comment:
@rtibbles Interested in your thoughts on this. I added this to detect when copying is finished so we can show the snackbars appropriately.